plugin-parser/mlops-plugin-parser.cc

```diff
@@ -125,19 +243,62 @@ _parse_json (JsonNode *node, mlsvc_json_type_e json_type, const gchar *app_info)
 
           if (!name || !model) {
             _E ("Failed to get name or model from MLSVC_JSON_MODEL.");
-            continue;
+            return FALSE;
           }
 
-          guint version;
-          bool active = (activate && g_ascii_strcasecmp (activate, "true") == 0);
+          if (event == MLSVC_PKGMGR_MDPARSER_PLUGIN_EVENT_TYPE_INSTALL) {
+            guint version;
+            bool active = (activate && g_ascii_strcasecmp (activate, "true") == 0);
 
-          ret = ml_agent_model_register (name, model, active, desc ? desc : "",
-              app_info ? app_info : "", &version);
+            ret = ml_agent_model_register (name, model, active,
+                desc ? desc : "", app_info ? app_info : "", &version);
 
-          if (ret == 0)
-            _I ("The model with name '%s' is registered as version '%u'.", name, version);
-          else
-            _E ("Failed to register the model with name '%s'.", name);
+            if (ret == 0)
+              _I ("The model with name '%s' is registered as version '%u'.", name, version);
+            else {
+              _E ("Failed to register the model with name '%s'.", name);
+              return FALSE;
+            }
+          } else if (event == MLSVC_PKGMGR_MDPARSER_PLUGIN_EVENT_TYPE_UNINSTALL) {
+            char *model_info = NULL;
+            ret = ml_agent_model_get_all (name, &model_info);
+
+            if (ret == 0) {
+              _uninstall_rpk (name, model_info, json_type);
+            } else {
+              _E ("Failed to get the model with name '%s'.", name);
+              return FALSE;
+            }
+
+          } else if (event == MLSVC_PKGMGR_MDPARSER_PLUGIN_EVENT_TYPE_UPGRADE) {
+            guint version;
+            bool active = (activate && g_ascii_strcasecmp (activate, "true") == 0);
+
+            ret = ml_agent_model_register (name, model, active,
+                desc ? desc : "", app_info ? app_info : "", &version);
+
+            if (ret == 0) {
+              char *model_info = NULL;
```

Comment:
ditto g_autofree gchar *model_info = NULL;

Reply:
thanks, i update it.